### PR TITLE
feature(reset password): users should be able to reset their passwords

### PR DESCRIPTION
**Pull request by ValerieRono**
#### What does this PR do?
A user should be able to reset his/her password when he/she forgets it, or for any other reason

#### Description of Tasks to be completed?
- create component where users can submit their email address and receive a reset password link
- create component where users can submit their new passwords and hence successfully reset their passwords
- write tests for said feature
- The user should receive **descriptive** validation errors in case of any.

#### How should this be manually tested?
- Clone the project: git clone https://github.com/andela/ah-centauri-frontend
- cd into the project folder
- checkout to branch `feature/164829161-password-reset `
- Install dependencies: yarn install
- Launch development server: yarn start
- enter `/reset` after url to access the reset password page
- enter your email address
- go to your inbox and click on the link provided which will redirect you
- enter new password, enter the password again to confirm they are a match

#### Any background context you want to provide?
N/A

#### What are the relevant Pivotal Tracker Stories?
[#164829161](https://www.pivotaltracker.com/story/show/164829161)

#### Screenshots
- get reset password link
image

- reset password form
image

### CODE REVIEW

File: src/components/resetPassword/resetPasswordForm.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
```
Comment (Mirriam-Maina) [inline src/components/resetPassword/resetPasswordForm.jsx:1]: @ValerieRono to increase your coverage, you can test this component using mount

File: src/actions/resetActions.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            }
+        });
+    };
+};
+
+export const resetPasswordService = ({data}) => {
+  return (dispatch) => {
+      api.user.resetPassword(data)
+      .then((response) => {
+          console.log(response.data)
```
Comment (bl4ck4ndbr0wn) [inline src/actions/resetActions.js]: Remove console.log

**Review by vino-jasuba — APPROVED** [review]
Great job on this @ValerieRono . 👍 Looks good to me.

File: src/components/resetPassword/resetPasswordForm.jsx (hunk 1 above)
Comment (ValerieRono) [reply src/components/resetPassword/resetPasswordForm.jsx:1]: The coverage report says this component has been tested 100% by way of testing the view that renders the form, I have added tests to the utils file

**Review by bl4ck4ndbr0wn — APPROVED** [review]
Great job on this @ValerieRono . 👍

**Review by Mirriam-Maina — COMMENTED** [review]
@ValerieRono good job done on this feature

- `Forgot password?` should be available as a link for the user to avoid prompting the user to edit the url with `/reset` manually

- Kindly rebase off develop again

**Comment by ValerieRono** [comment]
> @ValerieRono good job done on this feature
> 
> * `Forgot password?` should be available as a link for the user to avoid prompting the user to edit the url with `/reset` manually
> * Kindly rebase off develop again

feedback implemented

**Pull request merged**